test_common/harness/deviceInfo.cpp

```diff
@@ -29,44 +34,40 @@ static void *alloc_and_get_device_info( cl_device_id device, cl_device_info para
         throw std::runtime_error("clGetDeviceInfo failed\n");
     }
 
-    if (0 == size)
-        return NULL;
+    if (0 == size) return "";
 
-    auto buffer = new uint8_t[size];
+    std::vector<char> info(size);
 
-    if ((err = clGetDeviceInfo(device, param_name, size, buffer, NULL)) != CL_SUCCESS)
+    if ((err = clGetDeviceInfo(device, param_name, size, info.data(), NULL))
+        != CL_SUCCESS)
     {
-        delete [] buffer;
         throw std::runtime_error("clGetDeviceInfo failed\n");
     }
 
-    return buffer;
+    return std::string(info.begin(), info.end());
 }
 
 /* Determines if an extension is supported by a device. */
 int is_extension_available(cl_device_id device, const char *extensionName)
 {
-    char *extString = alloc_and_get_device_extensions_string(device);
-
-    BufferOwningPtr<char> extStringBuf(extString);
-
-    return strstr(extString, extensionName) != NULL;
+    std::string extString = get_device_extensions_string(device);
+    return extString.find(extensionName) != std::string::npos;
```

Comment:
Technically this is wrong (the function would report true for "cl_khr_prefix" on an implementation that supports "cl_khr_prefix_and_more"). We should really split the string on spaces and match against individual "words". Since this replicates the existing behaviour, I'm fine with doing this as a separate PR if you prefer.

Reply:
This occurred to me too but I couldn't think of any examples where this would be the case. Still, makes sense to be robust, so I've done this in this PR.